enterprise/frontend/src/metabase-enterprise/clean_up/CleanupCollectionModal/CleanupCollectionModal.tsx

```diff
@@ -0,0 +1,193 @@
+  type DateFilter,
+  getDateFilterLabel,
+  getDateFilterValue,
+  itemKeyFn,
+  getModalHeightCalc,
+} from "./utils";
+
+interface CleanupCollectionModalProps {
+  onClose: () => void;
+  params: { slug: string };
+}
+
+const _CleanupCollectionModal = ({
+  onClose: handleClose,
+  params: { slug },
+}: CleanupCollectionModalProps) => {
+  const dispatch = useDispatch();
+  const collectionId = Urls.extractCollectionId(slug);
+
+  // selection
+  const selection = useListSelect(itemKeyFn);
+
+  // pagination
+  const pagination = usePagination({ initialPage: 0, pageSize: 10 });
+
+  // sorting
+  const [sortOptions, setSortOptions] = useState<SortingOptions>({
+    sort_column: "name",
+    sort_direction: SortDirection.Asc,
+  });
+
+  const handleSortingChange = (sortingOpts: SortingOptions) => {
+    setSortOptions(sortingOpts);
+    pagination.resetPage();
+  };
+
+  // filters
+  const [dateFilter, setDateFilter] = useState<DateFilter>("six-months");
+  const handleChangeDateFilter = (nextDateFilter: DateFilter) => {
+    setDateFilter(nextDateFilter);
+    pagination.resetPage();
+    selection.clear();
+  };
+  const before_date = useMemo(() => {
+    return getDateFilterValue(dateFilter);
+  }, [dateFilter]);
+
+  const [recursiveFilter, setRecursiveFilter] = useState(false);
+  const handleChangeRecursiveFilter = (recursiveFilter: boolean) => {
+    setRecursiveFilter(recursiveFilter);
+    pagination.resetPage();
+    selection.clear();
+  };
+
+  // data
+  const {
+    data: staleItemsData,
+    isLoading,
+    error,
+  } = useListStaleCollectionItemsQuery(
+    collectionId
+      ? {
+          id: collectionId,
+          is_recursive: recursiveFilter,
+          before_date,
+          ...pagination.paginationFilters,
+          ...sortOptions,
+        }
+      : skipToken,
+    { refetchOnMountOrArgChange: true },
+  );
+
+  const itemsData = staleItemsData?.data;
+  const total = staleItemsData?.total ?? 0;
+  const items: StaleCollectionItem[] = useMemo(() => {
+    return (itemsData ?? []).map(item => Search.wrapEntity(item, dispatch));
+  }, [itemsData, dispatch]);
+
+  // selection cont.
+  const { getIsSelected } = selection;
+  const hasUnselected = useMemo(() => {
+    return items.some(item => !getIsSelected(item));
+  }, [getIsSelected, items]);
+
+  // pagination cont.
+  const { setTotal } = pagination;
+  useEffect(() => {
+    setTotal(total);
+  }, [setTotal, total]);
+
+  return (
+    <Modal.Root
+      opened
+      onClose={handleClose}
+      data-testid="cleanup-collection-modal"
+      size="70rem"
+    >
+      <Modal.Overlay />
+      <Modal.Content
+        className={CS.modalContent}
+        mih={isLoading ? `min(25rem, ${getModalHeightCalc("0px")})` : undefined}
+      >
+        <Modal.Header
+          px="2.5rem"
+          pt="2rem"
+          pb="1.5rem"
+          className={CS.modalHeader}
+        >
+          <Modal.Title fz="20px">{t`Select unused items to move to trash`}</Modal.Title>
```

Comment:
"Trash"?

Reply:
lowercase is consistent with other places in the UI. i can bring this up with product though.